src/osd/OSD.cc

```diff
@@ -5536,6 +5541,28 @@ void OSD::_send_boot()
   set_state(STATE_BOOTING);
 }
 
+std::string OSD::_collect_compression_algorithms()
+{
+  const auto& compression_algorithm_names = Compressor::compression_algorithm_names;
+  const auto& plugin_registry = cct->get_plugin_registry()->plugins;
+
+  if(0 == plugin_registry.size())
+   return {};
+
+  ostringstream os;
+  for_each(begin(compression_algorithm_names), end(compression_algorithm_names),
+  		  [&os, &plugin_registry](const auto& algorithm_name) {
+  		   if (plugin_registry.end() != plugin_registry.find(algorithm_name))
+  			os << algorithm_name << ',';
+  		  });
+  
+  string s(os.str());
+
+  s.erase(s.find_last_of(','));
```

Comment:
could just use
```c++
std::copy(begin(names), end(names), std::experimental::make_ostream_joiner(os, ", "));
```

Reply:
Agreed, are we feeling ok with things in std::experimental in general, though?

Reply:
we used to use facilities in the TS namespace before they are included by the std namespace. i think as long as it's part of standard, there is no reason not using it.